sparsesim/src/lib.rs

```diff
@@ -84,6 +84,11 @@ impl QuantumSim {
         self.rng.replace(StdRng::seed_from_u64(seed));
     }
 
+    pub fn get_next_rng(&mut self) -> StdRng {
+        let seed = self.rng.borrow_mut().gen::<u64>();
+        StdRng::seed_from_u64(seed)
+    }
+
```

Comment:
What about getting the rng instance out of the `RefCell` using `replace`:

```suggestion
    pub fn take_rng(&mut self) -> StdRng {
        self.rng.replace(StdRng::from_entropy())
    }
```